pkg/filecache/filecache.go

```diff
@@ -211,11 +212,15 @@ func (f *FileCacheProvider) Retrieve() (credentials.Value, error) {
 // otherwise fetching the credential from the underlying Provider and caching the results on disk
 // with an expiration time.
 func (f *FileCacheProvider) RetrieveWithContext(ctx context.Context) (credentials.Value, error) {
+	var quiet bool = viper.GetBool("quiet")
+
 	if !f.cachedCredential.Expired() && f.cachedCredential.HasKeys() {
 		// use the cached credential
 		return V2CredentialToV1Value(f.cachedCredential), nil
 	} else {
-		_, _ = fmt.Fprintf(os.Stderr, "No cached credential available.  Refreshing...\n")
+		if !quiet {
```

Comment:
i am not sure if we should be adding if else conditions and may be we should use klog or something similar and have these at default log level. If quiet we can set the klog to just print errors? Because if we add more logs, with this approach we need to add more if conditions. what do you think?

Reply:
I opted for the simplest option since only two strings were involved.
I can try to switch to klog, but it's a wider change.
Would you accept a PR that only migrates those 2 strings into klog with possible further refactoring of other logs? Because I don't feel comfortable testing all the log strings.

Reply:
I've just seen sirupsen/logrus is already used in the code. I'll try to use it instead of klog.
https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/cmd/aws-iam-authenticator/root.go#L28C23-L28C29

Reply:
I've updated the PR by using Logrus, which is the most obvious choice as it's used everywhere else in the code.

I should have seen that the actual issue was that filecache wasn't using Logrus.

I'm not sure if it can be considered a breaking change, but filecache logs will now include the level and timestamp from the Logrus formatter:

```
aws-iam-authenticator token --cache -i mycluster
INFO[2025-04-16T16:31:52+02:00] No cached credential available. Refreshing... 
INFO[2025-04-16T16:31:53+02:00] Updated cached credential                    
{"kind":"ExecCredential","apiVersion":"client.authentication.k8s.io/v1beta1","spec":{
```

Reply:
Hi @kmala,
Does my change from last month fit your recommendation ?
Thanks !